File: clojure/src/spoonacular_api/specs/inline_response_200_results.clj
(ns spoonacular-api.specs.inline-response-200-results
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def inline-response-200-results-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :calories) float?
   (ds/req :carbs) string?
   (ds/req :fat) string?
   (ds/req :image) string?
   (ds/req :imageType) string?
   (ds/req :protein) string?
   })

(def inline-response-200-results-spec
  (ds/spec
    {:name ::inline-response-200-results
     :spec inline-response-200-results-data}))

File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse200Results.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import java.math.BigDecimal

/**
 * 
 * @param id 
 * @param title 
 * @param calories 
 * @param carbs 
 * @param fat 
 * @param image 
 * @param imageType 
 * @param protein 
 */
case class InlineResponse200Results(id: Int,
                title: String,
                calories: BigDecimal,
                carbs: String,
                fat: String,
                image: String,
                imageType: String,
                protein: String
                )

object InlineResponse200Results {
    /**
     * Creates the codec for converting InlineResponse200Results from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse200Results] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse200Results] = deriveEncoder
}
